src/main/java/by/custom/utilcalculator/controller/MessageRouter.java

```diff
@@ -32,23 +33,25 @@ public SendMessage route(Update update) {
     }
 
     //we receive message from user, check it and handle it if it is ok here
-    public SendMessage getCheckInputMessageAndGetAnswer(Message message) {
+    private SendMessage getCheckInputMessageAndGetAnswer(Message message) {
         String usersMessage = message.getText();
         String userID = message.getChatId().toString();
+        UserProgress userProgress = UserStorageManager.getInstance().getUserProgress(userID);
```

Comment:
Please, move this logic to UserProgressManager. It's not responsibility of router to manage UserProgress

Reply:
Done.